site/content/extend/plugins/community-plugin-marketplace.md

```diff
@@ -0,0 +1,91 @@
+4. The plugin provides detailed usage documentation with at least one screenshot of the plugin in action, list of features and a development guide. This is typically a README file or a landing page on the web. The link to the documentation is set as `homepage_url` in the manifest. A great example is the [README of the GitHub plugin](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md).
+5. For every release a changelog has to be published, with a link recorded in the `release_notes_url` property of the `plugin.json` manifest.
+6. The plugin has to be out of Beta and be released with at least v1.0.0.
+7. All configuration is accessible via the UI of Mattermost.
+8. The plugin id defined in the manifest must not collide with the id of an existing plugin in the marketplace. It should follow [the naming convention](https://developers.mattermost.com/extend/plugins/manifest-reference/#id).
+
+**Technical Requirements (Checked by developers of the Toolkit or the Integrations team)**
+
+1. The plugin works for 60k concurrent connections and in a high availability environment. (There are currently no tools available to verify this property. Hence, it is checked via code review by a developer)
+2. The plugin logs important events on appropriate log levels to allow system admins to troubleshoot issues.
+
+**Security Requirements (Checked by a member of the security team)**
+
+1. The plugin does not expose a vulnerability.
+2. The plugins provided an email address or a username on the [Community Server](https://community.mattermost) used to report vulnerabilities in the future.
+
+**Functional Requirements (Checked by a QA tester)**
+
+1. The plugin must set a `min_server_version` in the manifest.
+2. The plugin must work on all Mattermost versions greater or equal then `min_server_version`.
+
+Please note, that Mattermost reserves the right to reject any plugin submission from the marketplace.
+
+Given that plugins are currently in Beta and rapid development is still happening, breaking changes especially for the webapp can happen. It is expected from plugin authors to keep up with these changes and fix bugs that may occur. Breaking changes will be mentioned in the "Important Upgrades Notes" section of the [Mattermost Changelog](https://docs.mattermost.com/administration/changelog.html).
+
+### Requirements for updating a community plugin to the Marketplace
+When a community plugin is updated, it must still fulfill the following checklist to be added to the Marketplace. This is checked by the four reviewers in the same way as when the plugin was added. The code review and security review should be performed against the `diff` of the last version in the Marketplace and the new version.
+
+The release also has to follow [semver](https://semver.org/). This specifically means for plugins:
+
+* If the plugin exposes an API for inter plugin communication, breaking changes to the API require a major version bump.
```

Comment:
There should be a hyphen between the words in "inter-plugin communication", but also it's not super clear what is meant by that. Does this refer to an API for multiple instance of a plugin to communicate with each other? Alternatively, does it refer to an API exposed for use by external clients? Does that include API endpoints intended for use by the plugin's extensions to the webapp? If all of these meanings are intended, I might simplify this to something along the lines of:
> If the plugin exposes an API, breaking changes to that API require a major version bump.

Reply:
Thanks for the feedback. :+1: I went with the wording from https://semver.org/:
> If the plugin exposes a public API, breaking changes to the API require a major version bump.